specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -2540,6 +2540,97 @@
         "url"
       ]
     },
+    "WebHookActivity": {
+      "x-ms-discriminator-value": "WebHook",
+      "description": "WebHook activity.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ControlActivity"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "WebHook activity properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WebHookActivityTypeProperties"
+        }
+      },
+      "required": [
+        "typeProperties"
+      ]
+    },
+    "WebHookActivityMethod": {
+      "description": "The list of HTTP methods supported by a WebHook activity.",
+      "type": "string",
+      "enum": [
+        "POST"
+      ],
+      "x-ms-enum": {
+        "name": "WebHookActivityMethod",
+        "modelAsString": true
+      }
+    },
+    "WebHookActivityAuthentication": {
+      "description": "WebHook activity authentication properties.",
+      "properties": {
+        "type": {
+          "description": "WebHook activity authentication (Basic/ClientCertificate/MSI)",
```

Comment:
You should make this one enum too

Reply:
This is the same as WebActivityAuthentication, copied over and renamed. I was going to reference that definition directly but didnt do so incase there would be some differences in the future. Should i still change it to enum?

Reply:
Changed this as per comments